provisionersdk/agent.go

```diff
@@ -18,6 +18,7 @@ Start-Process -FilePath $env:TEMP\sshd.exe -ArgumentList "agent" -PassThru`
 
 	linuxScript = `#!/usr/bin/env sh
 set -eux pipefail
+trap '[ $? -ne 0 ] && echo === Agent script exited with non-zero code. Sleeping infinitely to preserve logs... && sleep infinity' EXIT
```

Comment:
I'm a bit worried about `sleep infinity` compatibility. From a quick test it doesn't seem to work on macOS. It does seem supported in `busybox`, but e.g. in `toolbox` it doesn't (more common in Android world I think).

I wonder if it's be enough to sleep just a "long time", say something in the ballpark of 1-24h? I think anything about 15s should spare the Docker container from being erased.

PS. It doesn't hurt, but I don't think we need the non-zero exit check here since the `exec` at the end replaces this process, the trap should also go away. Up to you which you prefer, though.

Reply:
Wonderful catch, I'll make it a static duration (I agree 24h should be plenty).

The non-zero exit code can be removed with `exec`, another great catch!